core/templates/domain/skill/skill-backend-api.service.ts

```diff
@@ -16,127 +16,139 @@
-  'EDITABLE_SKILL_DATA_URL_TEMPLATE', 'SKILL_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, UrlInterpolationService,
-      EDITABLE_SKILL_DATA_URL_TEMPLATE, SKILL_DATA_URL_TEMPLATE) {
-    var _fetchSkill = function(skillId, successCallback, errorCallback) {
-      var skillDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_SKILL_DATA_URL_TEMPLATE, {
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { HttpClient } from '@angular/common/http';
+import { SkillChange } from 'domain/editor/undo_redo/ChangeObjectFactory';
+import { SkillDomainConstants } from 'domain/skill/skill-domain.constants';
+import { Skill, SkillBackendDict, SkillObjectFactory } from
+  'domain/skill/SkillObjectFactory';
+import { SkillSummaryBackendDict } from
+  'domain/skill/skill-summary-object.factory';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+interface FetchSkillBackendResponse {
+  'skill': SkillBackendDict;
+  'assigned_skill_topic_data_dict': {
+    [topicName: string]: string;
+  };
+  'grouped_skill_summaries': {
+    [topicName: string]: SkillSummaryBackendDict[];
+  }
+}
+
+interface FetchSkillResponse {
+  skill: Skill;
+  assignedSkillTopicData: {
+    [topicName: string]: string;
+  };
+  groupedSkillSummaries: {
+    [topicName: string]: SkillSummaryBackendDict[];
+  };
+}
+
+interface FetchMultiSkillsBackendResponse {
+  skills: SkillBackendDict[];
+}
+
+interface UpdateSkillBackendResponse {
+  skill: SkillBackendDict;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SkillBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private skillObjectFactory: SkillObjectFactory,
+    private urlInterpolationService: UrlInterpolationService) {}
+
+  fetchSkill(skillId: string): Promise<FetchSkillResponse> {
+    return new Promise((resolve, reject) => {
+      const skillDataUrl = this.urlInterpolationService.interpolateUrl(
+        SkillDomainConstants.EDITABLE_SKILL_DATA_URL_TEMPLATE, {
           skill_id: skillId
         });
 
-      $http.get(skillDataUrl).then(function(response) {
-        var assignedSkillTopicData = (
-          response.data.assigned_skill_topic_data_dict);
-        var skill = angular.copy(response.data.skill);
-        var groupedSkillSummaryDicts = angular.copy(
-          response.data.grouped_skill_summaries);
-        if (successCallback) {
-          successCallback({
-            skill: skill,
-            groupedSkillSummaries: groupedSkillSummaryDicts,
-            assignedSkillTopicData
+      this.http.get<FetchSkillBackendResponse>(skillDataUrl).toPromise()
+        .then(response => {
+          resolve({
+            skill: this.skillObjectFactory.createFromBackendDict(
+              response.skill),
+            assignedSkillTopicData: response.assigned_skill_topic_data_dict,
+            // TODO: Refactor this property to return SkillSummary domain
```

Comment:
This can't be done right now. Because the `groupedSkillSummaries` is passed in other directives like `questions-list` and there are many services do so and not all of them return domain objects.